src/redux/categories/categoriesSlice.js

```diff
@@ -0,0 +1,269 @@
+        'AVGO',
+        'CDNS',
+        'CSCO',
+        'CTSH',
+        'CRWD',
+        'DDOG',
+        'ENPH',
+        'FISV',
+        'FTNT',
+        'GFS',
+        'INTC',
+        'INTU',
+        'KLAC',
+        'LRCX',
+        'MRVL',
+        'MCHP',
+        'MU',
+        'MSFT',
+        'NVDA',
+        'NXPI',
+        'PANW',
+        'QCOM',
+        'SNPS',
+        'TXN',
+        'WDAY',
+        'ZM',
+        'ZS',
+      ],
+    },
+    {
+      category: 'Industrials',
+      stokes: [
+        'ADP',
+        'CTAS',
+        'CSX',
+        'FAST',
+        'HON',
+        'ODFL',
+        'PCAR',
+        'PAYX',
+        'VRSK',
+      ],
+    },
+    {
+      category: 'Consumer Cyclical',
+      stokes: [
+        'ABNB',
+        'AMZN',
+        'BKNG',
+        'CPRT',
+        'EBAY',
+        'JD',
+        'LCID',
+        'LULU',
+        'MAR',
+        'MELI',
+        'ORLY',
+        'PDD',
+        'RIVN',
+        'ROST',
+        'SBUX',
+        'TSLA',
+      ],
+    },
+    {
+      category: 'Healthcare',
+      stokes: [
+        'ALGN',
+        'AMGN',
+        'AZN',
+        'BIIB',
+        'DXCM',
+        'GILD',
+        'IDXX',
+        'ILMN',
+        'ISRG',
+        'MRNA',
+        'REGN',
+        'SGEN',
+        'VRTX',
+        'WBA',
+      ],
+    },
+    {
+      category: 'Utilities',
+      stokes: [
+        'AEP',
+        'CEG',
+        'EXC',
+        'XEL',
+      ],
+    },
+    {
+      category: 'Energy',
+      stokes: [
+        'BKR',
+        'FANG',
+      ],
+    },
+    {
+      category: 'Real Estate',
+      stokes: [
+        'CSGP',
+      ],
+    },
+    {
+      category: 'Consumer Defensive',
+      stokes: [
+        'COST',
+        'DLTR',
+        'KDP',
+        'KHC',
+        'MDLZ',
+        'MNST',
+        'PEP',
+      ],
+    },
+    {
+      category: 'Financial Services',
+      stokes: [
+        'PYPL',
+      ],
+    },
+  ],
+  stokes: [
+    'ADP',
+    'CTAS',
+    'CSX',
+    'FAST',
+    'HON',
+    'ODFL',
+    'PCAR',
+    'PAYX',
+    'VRSK',
+  ],
+  market: 'nasdaq_constituent',
+  category: 'Industrials',
+  companyName: '',
+  price: '',
+  image: '',
+  currentStoke: '',
+  status: 'idle',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    updateStokes: (state, action) => {
+      const { category } = action.payload;
+      let id;
+      for (let i = 0; i < state.categoriesList.length; i += 1) {
+        if (state.categoriesList[i].category === category) {
+          id = i;
+        }
+      }
+      const newStokes = state.categoriesList[id].stokes;
+      return {
+        ...state, stokes: newStokes, category, status: 'update',
+      };
+    },
+  },
+  extraReducers(builder) {
+    builder
+      .addCase(fetchStokes.fulfilled, (state, action) => {
+        const newCategoryList = [];
+        Object.keys(action.payload).forEach((cat) => {
```

Comment:
- [ ] **Good job so far 👏 👏** however, As suggested by the previous reviewer, It would be better to add a filter or a search input for your page where a user can type the name like **energy, technology**, etc, and get the filtered items on the home page 🙏 

   Kindly add this for the best user experience and to complete the project requirements 👍 👍 


   _I think this [youtube tutorial](https://www.youtube.com/watch?v=ZoayCCDHFiI) can help 😎😎😎_

Reply:
Hi. I think it is not clear what the instructions are. Can you explain it better? The fetch gives me a result that I filter by category to select the stokes of that category? Also I added a menu where the user can choose the market he wants. Can  you show me an example please?

Reply:
I don't see the point to do a filter. What I am doing is retrieving the name of the stoke, and with that name I made another fetch. Maybe that's the reason I don't need to filter again.